daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -228,16 +232,26 @@ object Hash {
   def assertFromString(s: String): Hash =
     data.assertRight(fromString(s))
 
+  def fromBytes(a: Array[Byte]): Either[String, Hash] =
+    Either.cond(
+      a.length == underlyingHashLength,
+      new Hash(a.clone()),
+      s"hash should have ${underlyingHashLength} bytes, found ${a.length}",
+    )
+
+  def assertFromBytes(a: Array[Byte]): Hash =
+    data.assertRight(fromBytes(a))
+
   def hashPrivateKey(s: String): Hash =
     builder(Purpose.PrivateKey).add(s).build
 
   // This function assumes that key is well typed, i.e. :
-  // 1 - `key.identifier` is the identifier for a template with a key of type τ
-  // 2 - `key.key` is a value of type τ
-  def hashContractKey(key: Node.GlobalKey): Hash =
+  // 1 - `templateId` is the identifier for a template with a key of type τ
+  // 2 - `key` is a value of type τ
+  def hashContractKey(templateId: Identifier, key: Value[Nothing]): Hash =
```

Comment:
`Value[Nothing]`? :thinking:

Reply:
It must not contain contract ids, hence the `Nothing`. This is following @remyhaemmerle-da 's code.